### ISSUE TITLE
Using <input type="file"/> throws SecurityError in Firefox

### ISSUE THREAD

**Issue by zdila** [facebook/react#7553]
**Do you want to request a _feature_ or report a _bug_?**
Bug.

**What is the current behavior?**
In development mode Firefox throws exception `SecurityError: The operation is insecure.`. It is thrown from `postMountWrapper` line `node.value = node.value;`.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://jsfiddle.net or similar (template: https://jsfiddle.net/reactjs/69z2wepo/).**

Demo:

``` javascript
render() {
  return <input type="file"/>;
}
```

**What is the expected behavior?**

No exception.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**
15.3.1. Not sure how did it behave in previous versions. FF is 48.0.

**Comment by zdila** [comment]
**Additional information:** It happens only if Firefox autofills the forms. If I use `<input type="file" autoComplete="off"/>` then I no more get this error.

**Comment by thanhthang20** [comment]
I got the same problem

**Comment by nhunzaker** [comment]
I'm familiar with this code and happy to work on a fix, but I'm struggling to reproduce the issue. Here's my test code:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
  </head>
  <body>
    <div id="container"></div>

    <script src="https://npmcdn.com/react@15.3.1/dist/react.js"></script>
    <script src="https://npmcdn.com/react-dom@15.3.1/dist/react-dom.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-core/5.8.24/browser.min.js"></script>

    <script type="text/babel">
      ReactDOM.render((
      <form method="POST" encType="multipart/form-data">
        <p><input name="photo" type="file" /></p>
        <p><input type="submit" /></p>
      </form>
      ), document.getElementById('container'))
    </script>
  </body>
</html>
```

Do you know if Firefox is autocompleting the file input itself?

**Comment by Aweary** [comment]
@zdila when you say "development mode", what are you referring to specifically?

**Comment by Aweary** [comment]
@zdila ping! I'd be interested in hearing more about how to reproduce this 😃

**Comment by zdila** [comment]
I was trying very hard to reproduce it again (when set `autoComplete="photo"`) but succeeded only once and now I can't reproduce it again. There must be some special conditions to get this :-(. 

Regarding _development mode_ I meant when `process.env.NODE_ENV !== "production"`

I can try again later but already spent 2 hours on it.

**Comment by zdila** [comment]
Note that when I was able to reproduce it I set FF debugger to stop on exceptions and after it stopped on `node.value = node.value` I inspected `node.value` and it was a pathname to a real file.

**Comment by Aweary** [comment]
@zdila were you ever able to reproduce this?

**Comment by zdila** [comment]
@Aweary since then I was no more working on that part, so no.

**Comment by aweary** [comment]
I'm going to go ahead close this since the issue is not reproducible anymore.

**Comment by jmzo** [comment]
I am experiencing this same bug. In hopes to help you reproduce... the steps I take are:

- open firefox
- using <input type='file' />, upload a file
- refresh page

Upon refresh I see...
<img width="1239" alt="screen shot 2017-05-31 at 7 43 22 am" src="https://cloud.githubusercontent.com/assets/4431979/26637694/07a4da8e-45d5-11e7-9c5e-24dd7455b99f.png">

Which links to...
<img width="637" alt="screen shot 2017-05-31 at 7 45 39 am" src="https://cloud.githubusercontent.com/assets/4431979/26637752/2d5e109c-45d5-11e7-9cb7-cf194b9ea445.png">

**Comment by jmzo** [comment]
I would add that @zdila s fix of using `autoComplete="off"` on the `<input type="file"/>` definitely fixed my issue, although you might still consider this a bug.

**Comment by stefanfisk** [comment]
I am also experiencing this right now.

**Comment by aweary** [comment]
If you can provide an example that consistently reproduces the issue using our [issue template](https://jsfiddle.net/84v837e9/), it would be really helpful @stefanfisk @jmzo!

**Comment by ryanmeisters** [comment]
problem for me was that I had `value="Upload..."` on the input. removing the `value` prop fixed it

**Comment by jarrod180** [comment]
I had `defaultValue="File"` , removing that fixed it for me.

**Comment by gaearon** [comment]
If you experience this please file a new issue with a complete reproducing example.
Commenting on a closed issue is not very helpful because we don't track them.
Thank you!
